examples/example-runner-cpu/src/main.rs

```diff
@@ -0,0 +1,54 @@
+use minifb::{Key, Window, WindowOptions};
+use rayon::prelude::*;
+use spirv_std::{vec2, Vec4};
+use std::time::Instant;
+
+fn color_u32_from_vec4(v: Vec4) -> u32 {
+    let convert = |f: f32| -> u32 { (f.min(1.0).max(0.0) * 255.0).round() as u32 };
+
+    convert(v.z()) | convert(v.y()) << 8 | convert(v.x()) << 16 | convert(v.w()) << 24
+}
+
+fn main() {
+    const WIDTH: usize = 1280;
+    const HEIGHT: usize = 720;
+
+    let mut window = Window::new(
+        "Rust GPU - CPU shader evaluation",
+        WIDTH,
+        HEIGHT,
+        WindowOptions::default(),
+    )
+    .expect("Window creation failed");
+
+    // Limit to max ~60 fps update rate
+    window.limit_update_rate(Some(std::time::Duration::from_micros(16600)));
+
+    let start_time = Instant::now();
+
+    let buffer = (0..WIDTH * HEIGHT)
+        .into_par_iter()
+        .map(|i| {
+            let screen_pos = vec2(
+                (i % WIDTH) as f32 / WIDTH as f32 * 2.0 - 1.0,
+                (i / HEIGHT) as f32 / HEIGHT as f32 * 2.0 - 1.0,
```

Comment:
I... thought this should be `(i / WIDTH)`, and I'm confused how this works at all (is the image shifted slightly compared to the GPU version?), but maybe it's just me just waking up.

Reply:
Just you waking up I think :) this is correct and works. Can likely be made cleaner in a separate functions or so maybe later, but does work

Reply:
It took me a while to wrap my head around this, but what happens is the following: The `i % WIDTH` generates an x coordinate in the range `0..WIDTH`, the `/ WIDTH` turns it into the range `0.0 .. 1.0`, the `* 2.0 - 1.0` then turns it into the range `-1.0 .. 1.0`. Same for the y coordinate.

Reply:
okay no I'm more awake now and it's *not* just me waking up, this *should* be `(i / WIDTH) as f32 / HEIGHT as f32 * 2.0 - 1.0`.

This PR's CPU code:

![cpu_original](https://user-images.githubusercontent.com/953151/97155649-f9883d80-1775-11eb-9748-0737e8174bb0.png)

GPU code:

![gpu](https://user-images.githubusercontent.com/953151/97155676-0016b500-1776-11eb-8e28-4f99bd422963.png)

This PR's CPU code but with the line changed to `i / WIDTH`:

![cpu_changed](https://user-images.githubusercontent.com/953151/97155710-0c027700-1776-11eb-87e0-91c3755d92b9.png)

it is indeed "the image shifted slightly compared to the GPU version"

Reply:
Hah awesome, I was overconfident that my code worked, well it did work just not correctly. Thanks!!